src/sys/socket/mod.rs

```diff
@@ -2009,12 +2009,116 @@ pub fn socketpair<T: Into<Option<SockProtocol>>>(
     unsafe { Ok((OwnedFd::from_raw_fd(fds[0]), OwnedFd::from_raw_fd(fds[1]))) }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct Backlog(i32);
+
+impl Backlog {
+    /// Sets the listen queue size to system `SOMAXCONN` value
+    #[cfg(target_os = "redox")]
+    pub const DEFAULT: Self = Self(128);
+    #[cfg(not(target_os = "redox"))]
+    pub const DEFAULT: Self = Self(libc::SOMAXCONN);
+    /// Sets the listen queue size to -1 for system supporting it
+    #[cfg(any(target_os = "linux", target_os = "freebsd"))]
+    pub const MAXALLOWABLE: Self = Self(-1);
+    /// Reasonable upper limit, while in theory we might be able
+    /// to set it to a higher boundary, comes with a risk
+    /// of resource exhaustion in return
+    pub const MAXVALUE: Self = Self(511);
```

Comment:
Is it really necessary to expose a `MAXVALUE` that is specified by us rather than the system?

Reply:
We might refine it a bit (e.g. linux/glibc which has SOMAXCONN to 4096, for this case we might go up to 65535).

Reply:
IMHO, we should not put a restriction that the OS does not have, so I kinda tend to remove this and use SOMAXCONN as the upper bound